backend/src/org/akvo/dash/import/csv.clj

```diff
@@ -32,7 +32,7 @@
   (format "CREATE %s %s (%s, %s)"
           (if temp? "TEMP TABLE" "TABLE")
           t-name
-          (if temp? "rnum serial" "rnum integer")
+          (if temp? "rnum serial primary key" "rnum integer primary key")
```

Comment:
Thanks for the explanation, that makes total sense. The integers are
already generated in the temp table

On Fri, Jun 3, 2016 at 9:21 PM Iván Perdomo <notifications@github.com>
wrote:

> In backend/src/org/akvo/dash/import/csv.clj
> <https://github.com/akvo/akvo-lumen/pull/210#discussion_r65762098>:
>
> > @@ -32,7 +32,7 @@
> >    (format "CREATE %s %s (%s, %s)"
> >            (if temp? "TEMP TABLE" "TABLE")
> >            t-name
> > -          (if temp? "rnum serial" "rnum integer")
> > +          (if temp? "rnum serial primary key" "rnum integer primary key")
>
> Serial is just a short-hand for having an integer column and a default
> value a call to a *sequence* function. In the case of the non-temp table,
> we're not going to add more rows, so we no need to generate more rnum
> values via call to the sequence function. It's a minor optimization to not
> have more dependent objects than needed.
>
> https://www.postgresql.org/docs/8.1/static/datatype.html#DATATYPE-SERIAL
>
> —
> You are receiving this because you modified the open/close state.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/akvo/akvo-lumen/pull/210/files/8e254a314ba5264c9ae9e9599a1d746a46a283b5#r65762098>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAGKn-rtOP5-d_KYoXrlmpkNf8X19RLiks5qIH7EgaJpZM4ItiPh>
> .
>